client/src/redux/actions/observations.js

```diff
@@ -22,6 +29,18 @@ export const addObservation = (userSession, observation) => async dispatch => {
   });
 };
 
+export const deleteObservations = userSession => async dispatch => {
+  const obs = await getObservations(userSession);
+  if (obs) {
+    const obsArray = JSON.parse(obs);
+    const numOfObservations = obsArray.length - 1;
+    deleteAllObservations(userSession).then(() => {
+      deleteSingleObservations(userSession, numOfObservations);
```

Comment:
just a small detail here, make sure to put these kinds of statements into try/catch blocks. you can also do `.catch` at the end of all the  `.then` logic. And remember to put a status code to send back after each action (I've been using 200)

Reply:
I'm personally confused about the status code stuff. Can you explain further or send me some documentation on this?

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200 

Since we're sending a request to change data we want to make sure not only that the request is made but also that it is successful. If the request is successful it will continue on to the `.then` where we can signal with a status code (in this case 200 for OK). otherwise it will go into the `.catch` where we'll get a log of the error. I think most dBs will send these codes automatically so you don't have to hard code them in redux doesn't do it on it's own so I think it's good practice to put these things in, in the case that if one day this request fails it will be quicker to isolate the issue. If we're not getting a status of 200 we know the data wasn't changed in the way we requested.

Reply:
And thinking about this further, I'm curious if Gaia does have a response that it sends that we could just log here instead of hard coding it in?

Reply:
I think the Blockstack API handles the HTTP-related stuff. We just focus on whether the `Promise` that the API call returns resolves or throws an error. I don't think we need to worry about handling status codes.

Reply:
cool, I'm still getting familiar with blockstack so I wasn't sure!